### PR TITLE
Encapsulate OutdatedVersionBanner element and rewrite metadata reducer in TypeScript

### PR DESCRIPTION
**Pull request by pedrottimark**
### Objectives

1. As a prerequisite to fix layout problems with banners:
    * Replace various data patterns with consistent encapsulation within components.
    * Therefore, simplify MainPage and minimize risk of re-rendering pages.
2. Improve type safety of reducers:
    * Replace classic createFetching abstraction for JavaScript with idiomatic TypeScript.
3. Prepare for follow-up to replace saga with thunk.

### Files changed

1. Edit src/Containers/MainPage/MainPage.tsx
    * Remove `metadata: { stale }` from Redux store selector.
    * Remove conditional rendering for `OutdatedVersionBanner` element.

2. Rename `VersionOutOfDate` as `OutdatedVersionBanner` for parallel naming convention
    * Encapsulate Redux store selector and conditional rendering.

3. Edit src/hooks/useMetadata.ts
    * Replace clumsy attempt at type safety with `useSelector` hook.

4. Rewrite ui/apps/platform/src/reducers/metadata.js in TypeScript
    * Define action object types.
    * Temporarily replace abstract `createFetchingActionTypes` and `createFetchingActions` calls with concrete action creator functions for compatibility with metadataSagas. See **Residue** item 1.
    * Replace invalid `null` with `metadataInitialState` which arbitrarily assumes release build.
    * Assign `metadata` with `Reducer<Metadata, MetadataAction>` type.
    * Rename `getMetadata` as `metadataSelector` selector.
    * Replace bogus `stale` property of metadata with auxiliary `outdatedVersion` reducer and `isOutdatedVersionSelector` selector.

### Residue

1. Replace metadataSagas with thunk in metadata reducer file.

## Checklist
- [x] Investigated and inspected CI test results
- [x] ~Unit test and regression tests added~

## Testing Performed

1. Temporarily add `console.log` in `metadata` reducer to see that actions still correspond to GET /v1/metadata requests and confirm `isEqual` works as expected.

2. Temporarily break comparison in `outdatedVersion` reducer to see that `OutdatedVersionBanner` component renders PatternFly `Banner` element.

### CODE REVIEW

File: ui/apps/platform/src/Containers/MainPage/OutdatedVersionBanner.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    window.location.reload();
+}
+
+function OutdatedVersionBanner(): ReactElement | null {
+    const isOutdatedVersion = useSelector(selectors.isOutdatedVersionSelector);
+
+    if (isOutdatedVersion) {
+        return (
+            <Banner className="pf-u-text-align-center" variant="warning">
+                It looks like this page is out of date and may not behave properly. Please{' '}
```
Comment (bradr5) [inline ui/apps/platform/src/Containers/MainPage/OutdatedVersionBanner.tsx:17]: Curious if you intentionally chose the curly bracket space `{' '}` here over a non-breaking space `&nbsp;`? Or is that just preference?

File: ui/apps/platform/src/reducers/metadata.ts [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    licenseStatus: 'VALID',
+    releaseBuild: true,
+    version: '', // response for request before authentication does not reveal version
+};
+
+const metadata: Reducer<Metadata, MetadataAction> = (
+    state = metadataInitialState,
+    action
+): Metadata => {
+    switch (action.type) {
```
Comment (bradr5) [inline ui/apps/platform/src/reducers/metadata.ts:54]: The benefits of typing this file out can also be noticed when highlighting properties such as `type` here

Screenshot 2023-02-21 at 11 16 28 AM

**Comment by bradr5** [comment]
I think this is a big improvement.

Only negative I want to point out is reading `OutdatedVersionBanner` initially made me think the component itself was outdated or old but maybe thats just me.

**Review by bradr5 — APPROVED** [review]

File: ui/apps/platform/src/Containers/MainPage/OutdatedVersionBanner.tsx (hunk 1 above)
Comment (pedrottimark) [reply ui/apps/platform/src/Containers/MainPage/OutdatedVersionBanner.tsx:17]: Good eyes. This is JSX notation for trailing space when line breaks. From prettier, I think.

**Pull request merged**